s3_bucket.go

```diff
@@ -175,16 +175,21 @@ func NewS3MultiPartBucketWithHTTPClient(client *http.Client, options S3Options)
 func (s *s3Bucket) String() string { return s.name }
 
 func (s *s3Bucket) Check(ctx context.Context) error {
-	input := &s3.GetBucketLocationInput{
+	input := &s3.HeadBucketInput{
 		Bucket: aws.String(s.name),
 	}
 
-	result, err := s.svc.GetBucketLocationWithContext(ctx, input, s3.WithNormalizeBucketLocation)
+	_, err := s.svc.HeadBucketWithContext(ctx, input)
+	// aside from a 404 Not Found error, HEAD bucket returns a 403
+	// Forbidden error. If the latter is the case, that is OK because
+	// we know the bucket exists and the given credentials may have
+	// access to a sub-bucket. See
+	// https://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketHEAD.html
+	// for more information.
 	if err != nil {
-		return errors.Wrap(err, "problem getting bucket location")
-	}
-	if *result.LocationConstraint != *s.svc.Client.Config.Region {
-		return errors.New("bucket does not exist in given region.")
+		if aerr, ok := err.(awserr.Error); ok && aerr.Code() == "NotFound" {
```

Comment:
nit: seems like there's probably something you can compare this to in the aws package (e.g. an error variable or a constant somewhere) rather than needing to use a string of your own construction.

Reply:
I thought so too... but the error that is said to be returned in the aws docs is not actually returned. The [HeadBucket documentation](https://docs.aws.amazon.com/sdk-for-go/api/service/s3/#S3.HeadBucket) states that `* ErrCodeNoSuchBucket "NoSuchBucket"` is returned upon error which is an S3 constant, but what actually is returned is "Not Found" which there is not constant for. See [here](https://docs.aws.amazon.com/sdk-for-go/api/service/s3/) for a list of constants. I did a grep for "Not Found" on the aws-sdk-go directory we have vendored without any luck.